shared/src/main/scala/org/adridadou/openlaw/parser/template/CompiledAgreement.scala

```diff
@@ -294,7 +294,10 @@ case class CompiledAgreement(
                 varType
                   .access(value, name, keys, executionResult)
                   .flatMap { option =>
-                    option.map(value => keysType.format(formatter, value, executionResult).map(_.toList)).getOrElse(Success(List()))
+                    option.map {
+                      case elements: OpenlawElements => Success(elements.underlying.toList)
```

Comment:
So I think this is a good way of handling it but we should also fix how we handle structure rendering.

I think the issue is that if the structure exists but the underlying info doesn't, then it creates an issue.

I think that the issue after refactoring is that something went wrong in how we get the underlying value.

Reply:
Ok, thanks for taking a look.  We can get this PR merged to fix the immediate issue and look more into the underlying change separately.